SBOM/spdx.md

```diff
@@ -0,0 +1,216 @@
+
+### Why SPDX?
+
+Initially, **alma-sbom** was implemented to generate SBOM documents in [OWASP CycloneDX](https://cyclonedx.org/) SBOM format, and as described in the [SBOM RFE](https://github.com/AlmaLinux/build-system-rfes/blob/master/SBOM/SBOM.md), we wanted to also extend its support to generate SBOM documents in the popular ISO-standarized format backed by the Linux Foundation, [SPDX](https://spdx.dev/).
+
+The benefit, the AlmaLinux OS community would be able to generate SBOM documents in two of the most popular SBOM standards available.
+
+## SPDX support overview
+
+### Changes to be made on alma-sbom
+
+We need to extend alma-sbom in different ways:
+
+#### 1. SPDX formatter
+
+Add an SPDX formatter in the same way we already have one for [CycloneDX](https://github.com/AlmaLinux/alma-sbom/blob/main/libsbom/cyclonedx.py)
+
+#### 2. Extending alma-sbom CLI
+
+Extend the command line options of alma-sbom in a way that we can generate SBOMs according to SPDX specification and in different output formats. Currently, we can generate SBOMs using CycloneDX into both JSON and XML formats, and we can generate SPDX SBOMs into:
+
+* JSON
+* XML
+* Tag/Value
+* YAML
+
+Generating SPDX SBOMs into different formats is currently provided by [spdx-tools](https://pypi.org/project/spdx-tools/), which alma-sbom will depend on.
+
+### SPDX SBOM data
+
+Both SPDX and CycloneDX SBOMs will share, mostly, the same data. There will be only changes in how the data is presented in SBOM records.
+To have an idea of the data that make up a SBOM record, you can check it in the [SBOM RFE](https://github.com/AlmaLinux/build-system-rfes/blob/master/SBOM/SBOM.md).
+
+Ideally, common data could be included as part of a DB record, this way, we could simplify how **alma-sbom** retrieves and formats data that is going to end up in SBOM records.
+
+In addition to that, we only add the data that is specific to SPDX specification. At the time of writing this RFE, the current version of the SPDX specification is [v2.3](https://spdx.github.io/spdx-spec/v2.3/).
+
+**TODO**: Describe SPDX-specific data (or fields)
+
+#### SPDX SBOM record of a package
+
+```javascript
+{
+    "SPDXID": "SPDXRef-424bb7a2-e4d7-472b-8260-23ec0feb7f07",
+    "creationInfo": {
+        "created": "2023-06-19T10:49:04.839885Z",
+        "creators": [
+            "Organization: AlmaLinux OS Foundation (cloud-infra@almalinux.org)",
+            "Tool: AlmaLinux Build System 0.1",
+            "Tool: alma-sbom 0.0.1",
+            "Tool: spdx-tools 0.0",
+            "Tool: Community Attestation Service (CAS) 1.0.3"
+        ]
+    },
+    "dataLicense": "CC0-1.0",
+    "name": "bash-5.1.8-6.el9_1",
+    "spdxVersion": "SPDX-2.3",
+    "documentNamespace": "https://almalinux.org/spdx",
```

Comment:
The URI does not have to be accessible. It is only intended to provide a unique ID. In many cases, the URI will point to a Web accessible document, but this should not be assumed to be the case.

If we follow this pattern: https://[CreatorWebsite]/[pathToSpdx]/[DocumentName]-[UUID]

Should we end up with something similar to: https://security.almalinux.org/spdx/bash-5.1.8-6.el9_1-424bb7a2-e4d7-472b-8260-23ec0feb7f07 ?

Reply:
This sounds good to me. This way all SBOMs have a unique id/namespace and we can make them downloadable at a later time (if it becomes necessary).

Reply:
Good, can you please incorporate such change into the PR?

Reply:
Yep, I just added a commit that addresses this issue